<launch>

<arg name="uavID" default="uav0" />
<arg name="fencePath" default = "$(find quadditch)/config/cage_5m_buffer_geofence.fen" />

<group ns="$(arg uavID)">

	<include file="$(find rosbridge_server)/launch/rosbridge_tcp.launch">
		<arg name = "bson_only_mode" value="True" />
	</include>


	<arg name="fcu_url" default="udp://:14551@" />
	<arg name="gcs_url" default="" />
	<arg name="tgt_system" default="1" />
	<arg name="tgt_component" default="1" />
	<arg name="log_output" default="screen" />
	<arg name="fcu_protocol" default="v2.0" />
	<arg name="respawn_mavros" default="false" />

	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/apm_pluginlists.yaml" />
		<arg name="config_yaml" value="$(find quadditch)/config/apm_config.yaml" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)" />
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
		<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
		<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
	</include>
		
	<rosparam param="fencePath" subst_value="True">$(arg fencePath)</rosparam>
	
	<node name="sitl" pkg = "quadditch" type="sim.sh" output="log"/>
	<node name="translator" pkg="quadditch" type="translator.py" output="screen"/>	
	<node name="uav_control" pkg="quadditch" type="uav_control.py" output="screen"/>
</group>

</launch>
